solver/include/solver/generic_butcher_rk.hpp

```diff
@@ -100,6 +100,296 @@ namespace lawson {
+    static constexpr value_t
+    value ( value_t x )
+    {
+      return static_cast<value_t>(2.)*x;
+    }
+  };
+
+  /**
+   * @class dT
+   * derivatives of Chebyshev polynomial: \f$\frac{\mathrm{d}T_N}{\mathrm{d}x}(x)=NU_{N-1}(x)\f$
+   * 
+   * @tparam N degree of polynomial
+   */
+  template <std::size_t N>
+  struct dT
+  {
+    template <typename value_t>
+    static constexpr value_t
+    value ( value_t x );
+  };
+
+  /**
+   * value of derivatives of Chebyshev polynomial: \f$\frac{\mathrm{d}T_N}{\mathrm{d}x}(x)\f$
+   * 
+   * @tparam value_t type of \f$x\f$
+   * @param x value where evaluate \f$\frac{\mathrm{d}T_N}{\mathrm{d}x}\f$
+   */
+  template <std::size_t N>
+  template <typename value_t>
+  constexpr value_t
+  dT<N>::value ( value_t x )
+  {
+    return N*U<N-1>::value(x);
+  }
+
+  template <>
+  struct dT<0>
+  {
+    template <typename value_t>
+    static constexpr value_t
+    value ( value_t x )
+    {
+      return static_cast<value_t>(0.);
+    }
+  };
+
+  /**
+   * @class ddT
+   * second derivative of Chebyshev polynomial: \f$\frac{\mathrm{d}^2T_N}{\mathrm{d}x^2}(x)=N\frac{NT_N(x) - xU_{N-1}(x)}{x^2 -1}\f$
+   * 
+   * @tparam N degree of polynomial
+   */
+  template <std::size_t N>
+  struct ddT
+  {
+    template <typename value_t>
+    static constexpr value_t
+    value ( value_t x );
+  };
+
+  /**
+   * value of derivative of Chebyshev polynomial: \f$\frac{\mathrm{d}T_N}{\mathrm{d}x}(x)\f$
+   * 
+   * @tparam value_t type of \f$x\f$
+   * @param x value where evaluate \f$\frac{\mathrm{d}T_N}{\mathrm{d}x}\f$
+   */
+  template <std::size_t N>
+  template <typename value_t>
+  constexpr value_t
+  ddT<N>::value ( value_t x )
+  {
+    return N*(N*T<N>::value(x) - x*U<N-1>::value(x))/(x*x - static_cast<value_t>(1.));
+  }
+
+  template <>
+  struct ddT<0>
+  {
+    template <typename value_t>
+    static constexpr value_t
+    value ( value_t x )
+    {
+      return static_cast<value_t>(0.);
+    }
+  };
+
+
+
+  /** @class explicit_rkc2
+   *  @brief define RKC2 with user defined number of stages
+   * 
+   *  @tparam _Nstages number of stages
+   *  @tparam value_t type of coefficients
+   */
+  template <std::size_t _N_stages, typename value_t=double>
+  struct explicit_rkc2
+  {
+    static constexpr bool is_embedded = false;
+    static constexpr std::size_t N_stages = _N_stages;
+    value_t w0;
+    value_t w1;
+
+    template <std::size_t J, typename T=value_t>
+    struct b
+    {
+      static constexpr value_t
+      value( value_t x )
+      {
+        return ddT<J>::value(x) / ( dT<J>::value(x)*dT<J>::value(x) );
+      }
+    };
+    template <typename T>
+    struct b<0,T>
+    {
+      static constexpr value_t
+      value( value_t x )
+      {
+        return b<2,value_t>::value(x);
+      }
+    };
+    template <typename T>
+    struct b<1,T>
+    {
+      static constexpr value_t
+      value( value_t x )
+      {
+        return b<2,value_t>::value(x);
+      }
+    };
```

Comment:
Why do you have this struct inside? You will not have to use it for other stuff?
A constexpr function would be better.

Reply:
Chebyshev polynomial can be use for other (futur) RKC methods, but these $b_j$ coefficients are only defined for this RKC2 method.